// @angular.studio/toolbar/fesm5/angular.studio-toolbar.js
import { __decorate } from 'tslib';
import { ɵɵdefineInjectable, Injectable, EventEmitter, Input, Output, Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatMenuModule } from '@angular/material/menu';

var ToolbarService = /** @class */ (function () {
    function ToolbarService() {
    }
    ToolbarService.ɵprov = ɵɵdefineInjectable({ factory: function ToolbarService_Factory() { return new ToolbarService(); }, token: ToolbarService, providedIn: "root" });
    ToolbarService = __decorate([
        Injectable({
            providedIn: 'root'
        })
    ], ToolbarService);
    return ToolbarService;
}());

var ToolbarComponent = /** @class */ (function () {
    function ToolbarComponent() {
        this.onClick = new EventEmitter();
    }
    __decorate([
        Input()
    ], ToolbarComponent.prototype, "config", void 0);
    __decorate([
        Output()
    ], ToolbarComponent.prototype, "onClick", void 0);
    ToolbarComponent = __decorate([
        Component({
            selector: 'toolbar',
            template: "\n\n        <div class=\"wrapper mat-elevation-z6\"\n             [style.background-color]=\"config.backgroundColor\">\n\n            <div class=\"left\">\n\n                <ng-content select=\"[left]\"></ng-content>\n\n            </div>\n\n            <div class=\"center\">\n\n                <ng-content select=\"[center]\"></ng-content>\n\n            </div>\n\n            <div class=\"right\">\n\n                <div class=\"menu\">\n\n                    <button mat-icon-button [matMenuTriggerFor]=\"menu\">\n\n                        <i [class]=\"config.menuIconClass\"\n                           [style.color]=\"config.menuIconColor\"\n                           [style.font-size]=\"config.menuIconFontSize\"></i>\n\n                    </button>\n\n                    <mat-menu #menu=\"matMenu\"\n                              [backdropClass]=\"config.menuClass\">\n\n                        <button *ngFor=\"let item of config.menuItems\"\n                                (click)=\"onClick.emit(item)\"\n                                mat-menu-item>\n\n                            <i *ngIf=\"item.iconClass\"\n                               [class]=\"item.iconClass\"\n                               style=\"margin-right: 10px\"\n                               [style.color]=\"item.iconColor\"></i>\n\n                            <span [style.color]=\"item.labelColor\">{{ item.label }}</span>\n\n                        </button>\n\n                    </mat-menu>\n\n                </div>\n\n            </div>\n\n        </div>\n\n    ",
            styles: [":host .wrapper{display:flex;align-items:center}:host .wrapper .center{flex:1}:host .wrapper .right .menu{display:flex;align-items:center;padding:5px}"]
        })
    ], ToolbarComponent);
    return ToolbarComponent;
}());

var ToolbarModule = /** @class */ (function () {
    function ToolbarModule() {
    }
    ToolbarModule = __decorate([
        NgModule({
            declarations: [ToolbarComponent],
            imports: [
                CommonModule,
                MatButtonModule,
                MatIconModule,
                MatMenuModule
            ],
            exports: [ToolbarComponent]
        })
    ], ToolbarModule);
    return ToolbarModule;
}());

var Toolbar = /** @class */ (function () {
    function Toolbar() {
        this.direction = 'horizontal';
    }
    return Toolbar;
}());

/*
 * Public API Surface of toolbar
 */

/**
 * Generated bundle index. Do not edit.
 */

export { Toolbar, ToolbarComponent, ToolbarModule, ToolbarService };
