// @angular.studio/toolbar/fesm2015/angular.studio-toolbar.js
import { __decorate } from 'tslib';
import { ɵɵdefineInjectable, Injectable, EventEmitter, Input, Output, Component, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatMenuModule } from '@angular/material/menu';

let ToolbarService = class ToolbarService {
    constructor() {
    }
};
ToolbarService.ɵprov = ɵɵdefineInjectable({ factory: function ToolbarService_Factory() { return new ToolbarService(); }, token: ToolbarService, providedIn: "root" });
ToolbarService = __decorate([
    Injectable({
        providedIn: 'root'
    })
], ToolbarService);

let ToolbarComponent = class ToolbarComponent {
    constructor() {
        this.onClick = new EventEmitter();
    }
};
__decorate([
    Input()
], ToolbarComponent.prototype, "config", void 0);
__decorate([
    Output()
], ToolbarComponent.prototype, "onClick", void 0);
ToolbarComponent = __decorate([
    Component({
        selector: 'toolbar',
        template: `

        <div class="wrapper mat-elevation-z6"
             [style.background-color]="config.backgroundColor">

            <div class="left">

                <ng-content select="[left]"></ng-content>

            </div>

            <div class="center">

                <ng-content select="[center]"></ng-content>

            </div>

            <div class="right">

                <div class="menu">

                    <button mat-icon-button [matMenuTriggerFor]="menu">

                        <i [class]="config.menuIconClass"
                           [style.color]="config.menuIconColor"
                           [style.font-size]="config.menuIconFontSize"></i>

                    </button>

                    <mat-menu #menu="matMenu"
                              [backdropClass]="config.menuClass">

                        <button *ngFor="let item of config.menuItems"
                                (click)="onClick.emit(item)"
                                mat-menu-item>

                            <i *ngIf="item.iconClass"
                               [class]="item.iconClass"
                               style="margin-right: 10px"
                               [style.color]="item.iconColor"></i>

                            <span [style.color]="item.labelColor">{{ item.label }}</span>

                        </button>

                    </mat-menu>

                </div>

            </div>

        </div>

    `,
        styles: [":host .wrapper{display:flex;align-items:center}:host .wrapper .center{flex:1}:host .wrapper .right .menu{display:flex;align-items:center;padding:5px}"]
    })
], ToolbarComponent);

let ToolbarModule = class ToolbarModule {
};
ToolbarModule = __decorate([
    NgModule({
        declarations: [ToolbarComponent],
        imports: [
            CommonModule,
            MatButtonModule,
            MatIconModule,
            MatMenuModule
        ],
        exports: [ToolbarComponent]
    })
], ToolbarModule);

class Toolbar {
    constructor() {
        this.direction = 'horizontal';
    }
}

/*
 * Public API Surface of toolbar
 */

/**
 * Generated bundle index. Do not edit.
 */

export { Toolbar, ToolbarComponent, ToolbarModule, ToolbarService };
